src/core/menu/mod.rs

```diff
@@ -0,0 +1,68 @@
+mod credits;
+
+use bevy::prelude::*;
+
+use super::CoreState;
+use crate::ui::*;
+
+pub(super) fn plugin(app: &mut App) {
+    // Setup state
+    app.add_sub_state::<MenuState>();
+    // Add state scoped entities for UI cleanup
+    app.enable_state_scoped_entities::<MenuState>();
+    // Print state transitions in debug builds
+    #[cfg(feature = "dev")]
+    app.add_systems(
+        Update,
+        bevy::dev_tools::states::log_transitions::<MenuState>,
+    );
+
+    // Setup, update, teardown
+    app.add_systems(OnEnter(MenuState::Main), setup);
+    app.add_systems(Update, update.run_if(in_state(MenuState::Main)));
+
+    // Sub plugins
+    app.add_plugins(credits::plugin);
+}
+
+#[derive(SubStates, Debug, PartialEq, Hash, Eq, Clone, Default)]
+#[source(CoreState = CoreState::Menu)]
+enum MenuState {
+    #[default]
+    Main,
+    Credits,
+}
+
+fn setup(mut commands: Commands) {
```

Comment:
Non-specific system names like "set up" and "update" are confusing, and encourage putting everything in one system instead of splitting into multiple systems. Fine to keep it this way for this PR, though.

Reply:
Yes, but overly specific names without context would be weird too. Adding some actual game logic would show both sides

Reply:
I added a comment for game logic, hope we can iterate on this further though